src/RazorSdk/SourceGenerators/MetadataReferenceExtensions.cs

```diff
@@ -14,18 +15,15 @@ internal static class MetadataReferenceExtensions
 
             if (symbol is IAssemblySymbol assemblySymbol)
             {
-                foreach (var module in assemblySymbol.Modules)
+                var module = assemblySymbol.Modules.First();
```

Comment:
The `Modules` collection likely isn't guaranteed to be ordered hence this cache may not be correct across invocations. I would instead do `SingleOrDefault` and just not cache in the multi-module case.